### PR TITLE
Support compressed images and RGBA data in Image class

### PR DESCRIPTION
**Pull request by luca-della-vedova**
# 🎉 New feature

Work towards #344 , specifically https://github.com/gazebosim/gz-common/issues/344#issuecomment-1141869020 to allow loading embedded textures.
Main user of this API would be gz-rendering, discussion on downstream API taking place in the [gz-rendering PR](https://github.com/gazebosim/gz-rendering/pull/642) 

## Summary
This PR adds support for loading compressed images from in memory binary blobs, this is achieved through a new `SetFromCompressedData` function that takes as an input a byte array and a format and loads the image internally. For now only PNG is supported but other formats can be added trivially by adding a new case to the if statement.
Since images are provided as binary blobs, height and width are not known when the function is called. An alternative approach could be to use `SetFromData` with the byte buffer being passed as an image of `1 * len(buffer)` size, but I thought that wasn't too clean so went for another function.

It also adds a `RGBAData` function that returns the image in RGBA8 format. The main use case for this is that loading a PNG without an alpha channel through `FreeImage` would return an RGB8 24 bit image. This format might be tricky to use for downstream users because it's not aligned to the 32 bit boundary. This is specifically an issue in OGRE2 which is unable to transfer 24 bit textures to GPU (ref [here](https://github.com/OGRECave/ogre-next/blob/master/OgreMain/include/OgrePixelFormatGpu.h#L187-L191)).
Hence the convenience function is added for downstream users to make sure the image is always in a 32 bit RGBA8 format.  

Note, I'm personally not a big fan of the API for the `Data` functions where a user has to remember to manually delete[] the pointer they passed as soon as they are done with it, it would probably be safer to either return a `std::vector` of bytes (at the risk of expensive copies happening if people are not careful) or a `std::shared_ptr` to a `std::vector` of bytes, which would make sure copies only increase the reference count and memory is still correctly deallocated.
However having this API would have required changing the behavior of the `DataImpl` function that is used in other parts of the library so I didn't quite know if this was a good idea.

## Test it
Unit tests have been added, manual test is significantly more complex since it requires using the branches in #344 and the [gz-rendering PR](https://github.com/gazebosim/gz-rendering/pull/642), having a GLB asset and making sure the asset is correctly displayed.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: graphics/src/Image.cc [hunk 1]
```diff
@@ -242,6 +242,28 @@ void Image::SetFromData(const unsigned char *_data,
...
+                                  unsigned int _size,
+                                  Image::PixelFormatType _format)
+{
+  if (this->dataPtr->bitmap)
+    FreeImage_Unload(this->dataPtr->bitmap);
+  this->dataPtr->bitmap = nullptr;
+
+  if (_format == COMPRESSED_PNG)
+  {
+    FIMEMORY *fiMem = FreeImage_OpenMemory(const_cast<unsigned char*>(_data), _size);
```
Comment (mjcarroll) [inline graphics/src/Image.cc:256]: `const_cast` is a little concerning here.  Can we change the API to reflect that `const`-ness may be violated?

Comment (luca-della-vedova) [reply graphics/src/Image.cc]: Done https://github.com/gazebosim/gz-common/pull/368/commits/8054b0d78433da3de936ffc9b81d87295207a26f

**Comment by mjcarroll** [comment]
> However having this API would have required changing the behavior of the DataImpl function that is used in other parts of the library so I didn't quite know if this was a good idea.

I agree completely.  Fixing this API has been on my someday/maybe list for a while.  While it is out of the scope for this PR, if anyone on your side has some spare cycles, I would really appreciate a cleaner API before Garden.

**Comment by mjcarroll** [comment]
Windows failing is expected.

**Comment by luca-della-vedova** [comment]
> > However having this API would have required changing the behavior of the DataImpl function that is used in other parts of the library so I didn't quite know if this was a good idea.
> 
> I agree completely. Fixing this API has been on my someday/maybe list for a while. While it is out of the scope for this PR, if anyone on your side has some spare cycles, I would really appreciate a cleaner API before Garden.

I'm happy to look at it, I ticketed an issue https://github.com/gazebosim/gz-common/issues/369 and can start working on it if the idea sounds reasonable to you.

**Comment by mjcarroll** [comment]
> I'm happy to look at it, I ticketed an issue https://github.com/gazebosim/gz-common/issues/369 and can start working on it if the idea sounds reasonable to you.

Great, thanks!  I think improving this API will both make it more reasonable for users as well as make it generally more efficient as we are doing a lot of reallocations and copies.

**Review by mjcarroll — APPROVED** [review]
LGTM, I just added two more quick test cases so we can get :100: coverage on the diff.

**Pull request merged**